[2023-02-24T21:41:36.002] [ERROR] fileWarn - ReferenceError: usuariosDao is not defined
    at Strategy._verify (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Middlewares/passport.js:40:26)
    at Strategy.authenticate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\passport-local\lib\strategy.js:88:12)
    at attempt (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\passport\lib\middleware\authenticate.js:369:16)
    at authenticate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\passport\lib\middleware\authenticate.js:370:7)
    at Layer.handle [as handle_request] (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\route.js:144:13)
    at Route.dispatch (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\route.js:114:3)
    at Layer.handle [as handle_request] (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\layer.js:95:5)
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\index.js:284:15
    at Function.process_params (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\index.js:346:12)
[2023-02-27T15:11:04.227] [ERROR] fileError - TypeError: item.save is not a function
    at CarritoMongoDb.deleteById (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Containers/container.mongodb.js:65:18)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Routes/carritos.routes.js:51:9
[2023-02-27T15:51:30.709] [ERROR] fileWarn - TypeError: Cannot read properties of undefined (reading 'isDeleted')
    at file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Routes/carritos.routes.js:32:25
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2023-03-01T15:40:28.760] [ERROR] fileError - Error: productos validation failed: nombre: Path `nombre` is required., urlImagen: Path `urlImagen` is required., precio: Path `precio` is required.
    at ValidationError.inspect (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\error\validation.js:50:26)
    at formatValue (node:internal/util/inspect:780:19)
    at inspect (node:internal/util/inspect:345:10)
    at formatWithOptionsInternal (node:internal/util/inspect:2165:40)
    at Object.format (node:internal/util/inspect:2019:10)
    at basicLayout (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\layouts.js:65:52)
    at app (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\appenders\file.js:98:23)
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\appenders\logLevelFilter.js:10:7
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\log4js.js:43:5
    at Array.forEach (<anonymous>) {
  errors: {
    nombre: ValidatorError: Path `nombre` is required.
        at validate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1346:13)
        at SchemaString.SchemaType.doValidate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1330:7)
        at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\document.js:2905:18
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      properties: [Object],
      kind: 'required',
      path: 'nombre',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose:validatorError)]: true
    },
    urlImagen: ValidatorError: Path `urlImagen` is required.
        at validate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1346:13)
        at SchemaString.SchemaType.doValidate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1330:7)
        at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\document.js:2905:18
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      properties: [Object],
      kind: 'required',
      path: 'urlImagen',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose:validatorError)]: true
    },
    precio: ValidatorError: Path `precio` is required.
        at validate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1346:13)
        at SchemaNumber.SchemaType.doValidate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1330:7)
        at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\document.js:2905:18
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      properties: [Object],
      kind: 'required',
      path: 'precio',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose:validatorError)]: true
    }
  },
  _message: 'productos validation failed'
}
[2023-03-01T21:55:49.665] [ERROR] fileWarn - ReferenceError: usuariosDao is not defined
    at Strategy.loginPassport [as _verify] (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/usuarios.controller.js:42:26)
    at Strategy.authenticate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\passport-local\lib\strategy.js:88:12)
    at attempt (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\passport\lib\middleware\authenticate.js:369:16)
    at authenticate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\passport\lib\middleware\authenticate.js:370:7)
    at Layer.handle [as handle_request] (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\route.js:144:13)
    at Route.dispatch (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\route.js:114:3)
    at Layer.handle [as handle_request] (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\layer.js:95:5)
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\index.js:284:15
    at Function.process_params (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\router\index.js:346:12)
[2023-03-01T21:56:30.847] [ERROR] fileWarn - ReferenceError: comparar is not defined
    at Strategy.loginPassport [as _verify] (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/usuarios.controller.js:48:37)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2023-03-01T22:28:16.358] [ERROR] fileWarn - ReferenceError: transport is not defined
    at Strategy.signupPassport [as _verify] (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/usuarios.controller.js:108:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2023-03-01T22:53:43.543] [ERROR] fileWarn - ReferenceError: actualizarProducto is not defined
    at updateProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:69:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2023-03-02T14:18:17.588] [ERROR] fileWarn - TypeError: Cannot read properties of undefined (reading 'timestamp')
    at new CarritosDto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/DTOs/carritos.dto.js:3:34)
    at RepositorioCarritos.getById (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Repository/carritos.repository.js:15:16)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async obtenerCarritoPorId (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Service/carritos.service.js:16:21)
    at async saveProductInCart (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/carritos.controller.js:71:26)
[2023-03-02T14:19:38.101] [ERROR] fileWarn - TypeError: Cannot read properties of undefined (reading 'timestamp')
    at new CarritosDto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/DTOs/carritos.dto.js:3:34)
    at RepositorioCarritos.getById (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Repository/carritos.repository.js:15:16)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async obtenerCarritoPorId (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Service/carritos.service.js:16:21)
    at async saveProductInCart (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/carritos.controller.js:71:26)
[2023-03-04T15:08:04.363] [ERROR] fileError - Error: productos validation failed: urlImagen: Path `urlImagen` is required., precio: Path `precio` is required.
    at ValidationError.inspect (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\error\validation.js:50:26)
    at formatValue (node:internal/util/inspect:780:19)
    at inspect (node:internal/util/inspect:345:10)
    at formatWithOptionsInternal (node:internal/util/inspect:2165:40)
    at Object.format (node:internal/util/inspect:2019:10)
    at basicLayout (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\layouts.js:65:52)
    at app (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\appenders\file.js:98:23)
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\appenders\logLevelFilter.js:10:7
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\log4js.js:43:5
    at Array.forEach (<anonymous>) {
  errors: {
    urlImagen: ValidatorError: Path `urlImagen` is required.
        at validate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1346:13)
        at SchemaString.SchemaType.doValidate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1330:7)
        at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\document.js:2905:18
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      properties: [Object],
      kind: 'required',
      path: 'urlImagen',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose:validatorError)]: true
    },
    precio: ValidatorError: Path `precio` is required.
        at validate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1346:13)
        at SchemaNumber.SchemaType.doValidate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1330:7)
        at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\document.js:2905:18
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      properties: [Object],
      kind: 'required',
      path: 'precio',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose:validatorError)]: true
    }
  },
  _message: 'productos validation failed'
}
[2023-03-04T15:09:27.175] [ERROR] fileError - Error: productos validation failed: urlImagen: Path `urlImagen` is required., precio: Path `precio` is required.
    at ValidationError.inspect (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\error\validation.js:50:26)
    at formatValue (node:internal/util/inspect:780:19)
    at inspect (node:internal/util/inspect:345:10)
    at formatWithOptionsInternal (node:internal/util/inspect:2165:40)
    at Object.format (node:internal/util/inspect:2019:10)
    at basicLayout (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\layouts.js:65:52)
    at app (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\appenders\file.js:98:23)
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\appenders\logLevelFilter.js:10:7
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\log4js\lib\log4js.js:43:5
    at Array.forEach (<anonymous>) {
  errors: {
    urlImagen: ValidatorError: Path `urlImagen` is required.
        at validate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1346:13)
        at SchemaString.SchemaType.doValidate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1330:7)
        at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\document.js:2905:18
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      properties: [Object],
      kind: 'required',
      path: 'urlImagen',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose:validatorError)]: true
    },
    precio: ValidatorError: Path `precio` is required.
        at validate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1346:13)
        at SchemaNumber.SchemaType.doValidate (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\schematype.js:1330:7)
        at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongoose\lib\document.js:2905:18
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      properties: [Object],
      kind: 'required',
      path: 'precio',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose:validatorError)]: true
    }
  },
  _message: 'productos validation failed'
}
[2023-03-04T15:47:22.998] [ERROR] fileError - MongoServerError: E11000 duplicate key error collection: Ecommerce.productos index: codigo_1 dup key: { codigo: "003" }
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongodb\lib\operations\insert.js:53:33
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongodb\lib\cmap\connection_pool.js:333:25
    at C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongodb\lib\sdam\server.js:210:17
    at handleOperationResult (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongodb\lib\sdam\server.js:326:20)
    at Connection.onMessage (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongodb\lib\cmap\connection.js:242:9)
    at MessageStream.<anonymous> (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongodb\lib\cmap\connection.js:61:60)
    at MessageStream.emit (node:events:520:28)
    at processIncomingData (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongodb\lib\cmap\message_stream.js:125:16)
    at MessageStream._write (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\mongodb\lib\cmap\message_stream.js:33:9)
    at writeOrBuffer (node:internal/streams/writable:389:12) {
  index: 0,
  code: 11000,
  keyPattern: { codigo: 1 },
  keyValue: { codigo: '003' },
  [Symbol(errorLabels)]: Set(0) {}
}
[2023-03-11T22:41:36.281] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T22:46:43.493] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T22:49:11.492] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T22:59:21.524] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T23:02:11.711] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T23:04:11.925] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T23:05:15.760] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T23:05:27.545] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T23:11:49.923] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T23:16:27.678] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T23:22:23.382] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-11T23:33:39.985] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
[2023-03-14T19:32:39.131] [ERROR] fileWarn - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:794:10)
    at ServerResponse.send (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:174:12)
    at ServerResponse.json (C:\Users\mauri\Desktop\c32125_PFbackend\backend\node_modules\express\lib\response.js:278:15)
    at getProducto (file:///C:/Users/mauri/Desktop/c32125_PFbackend/backend/Controller/productos.controller.js:14:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
